test/test_autograd.py

```diff
@@ -4499,6 +4502,105 @@ def fn(*inputs):
 
             setattr(TestAutogradDeviceType, test_name, do_test)
 
+class TestAutogradComplex(TestCase):
+    # remove this test after gradcheck support is added for non-holomoprphic functions
+    def test_real(self):
+        x = torch.randn(10, dtype=torch.cdouble, requires_grad=True)
+        x.real.sum().backward()
+        self.assertEqual(x.grad, torch.ones_like(x))
+
+    # remove this test after gradcheck support is added for non-holomoprphic functions
+    def test_imag(self):
+        x = torch.randn(10, dtype=torch.cdouble, requires_grad=True)
+        x.imag.sum().backward()
+        self.assertEqual(x.grad, -1j * torch.ones_like(x))
+
+    # remove this test after gradcheck support is added for non-holomoprphic functions
+    def test_view_as_real(self):
+        x = torch.randn(10, dtype=torch.cdouble, requires_grad=True)
+        torch.view_as_real(x).sum().backward()
+        self.assertEqual(x.grad, torch.full_like(x, 1 - 1j))
+
+    # remove this test after gradcheck support is added for non-holomoprphic functions
+    def test_view_as_complex(self):
+        x = torch.randn(10, 2, dtype=torch.double, requires_grad=True)
+        torch.view_as_complex(x).sum().backward()
+        self.assertEqual(x.grad, torch.tensor([1, 0], dtype=torch.double).expand_as(x))
+
+    def test_view_func_for_complex_views(self):
```

Comment:
Is there a test for in-place operations on .real and .imag? No, right, because that will be part of the setter PR?

Reply:
yeah that's correct